main_test.go

```diff
@@ -37,3 +37,100 @@ func TestBuildExporter(t *testing.T) {
 
 	buildExporter(opts)
 }
+
+func TestBuildURI(t *testing.T) {
+	const newUser = "xxx"
+	const newPass = "yyy"
+
+	const originalBareURI = "127.0.0.1"
+	const originalAuthURI = "usr:pwd@127.0.0.1"
+
+	const originalPrefixBareURI = "mongodb://127.0.0.1"
+	const originalPrefixAuthURI = "mongodb://usr:pwd@127.0.0.1"
+	const changedPrefixAuthURI = "mongodb://xxx:yyy@127.0.0.1"
+
+	var newUri string
+	resetNewUri := func() {
+		newUri = ""
+	}
+
+	t.Log("\nuri with prefix and auth, and auth supplied in opt.User/Password")
+	newUri = buildURI(originalPrefixAuthURI, newUser, newPass)
+	t.Logf("Origin: %s", originalPrefixAuthURI)
+	t.Logf("Expect: %s", originalPrefixAuthURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != originalPrefixAuthURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with prefix and auth, no auth supplied in opt.User/Password")
+	newUri = buildURI(originalPrefixAuthURI, "", "")
+	t.Logf("Origin: %s", originalPrefixAuthURI)
+	t.Logf("Expect: %s", originalPrefixAuthURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != originalPrefixAuthURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with no prefix and auth, and auth supplied in opt.User/Password")
+	newUri = buildURI(originalAuthURI, newUser, newPass)
+	t.Logf("Origin: %s", originalAuthURI)
+	t.Logf("Expect: %s", originalAuthURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != originalAuthURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with no prefix and auth, no auth supplied in opt.User/Password")
+	newUri = buildURI(originalAuthURI, "", "")
+	t.Logf("Origin: %s", originalAuthURI)
+	t.Logf("Expect: %s", originalAuthURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != originalAuthURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with prefix and no auth, and auth supplied in opt.User/Password")
+	newUri = buildURI(originalPrefixBareURI, newUser, newPass)
+	t.Logf("Origin: %s", originalPrefixBareURI)
+	t.Logf("Expect: %s", changedPrefixAuthURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != changedPrefixAuthURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with prefix and no auth, no auth supplied in opt.User/Password")
+	newUri = buildURI(originalPrefixBareURI, "", "")
+	t.Logf("Origin: %s", originalPrefixBareURI)
+	t.Logf("Expect: %s", originalPrefixBareURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != originalPrefixBareURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with no prefix and no auth, and auth supplied in opt.User/Password")
+	newUri = buildURI(originalBareURI, newUser, newPass)
+	t.Logf("Origin: %s", originalBareURI)
+	t.Logf("Expect: %s", changedPrefixAuthURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != changedPrefixAuthURI {
+		t.Fail()
+	}
+	resetNewUri()
+
+	t.Log("\nuri with no prefix and no auth, no auth supplied in opt.User/Password")
+	newUri = buildURI(originalBareURI, "", "")
+	t.Logf("Origin: %s", originalBareURI)
+	t.Logf("Expect: %s", originalBareURI)
+	t.Logf("Result: %s", newUri)
+	if newUri != originalBareURI {
```

Comment:
```suggestion
	if newUri != originalPrefixBareURI {
```
This is the expected behavior.

Reply:
i have reorgnized the test code, https://github.com/percona/mongodb_exporter/pull/560/commits/24987e44cf7c10d7678effa7d9fdb27f5fb88016

could you review it again, thanks